mage_ai/api/resources/LogResource.py

```diff
@@ -7,8 +7,63 @@
 from mage_ai.orchestration.db.models.schedules import BlockRun, PipelineRun, PipelineSchedule
 from sqlalchemy.orm import aliased
 from typing import Dict, List
+import json
+import re
+import time
 
 MAX_LOG_FILES = 20
+TIMESTAMP_REGEX = re.compile(r'([0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}) (.+)')
+
+
+def is_json_string(string):
+    if not string:
+        return False
+
+    try:
+        json.loads(string)
+    except ValueError:
+        return False
+
+    return True
+
+
+def initialize_log(log):
+    content = log['content']
+    match = re.match(TIMESTAMP_REGEX, content)
+
+    datetime = match.group(1) if match else None
+    data_raw = match.group(2) if match else None
+
+    data = {}
+    if data_raw and is_json_string(data_raw):
+        data = json.loads(data_raw)
+
+    return {
+        **log,
+        'created_at': datetime,
+        'data': data,
+    }
+
+
+def initialize_logs(log) -> List[Dict]:
```

Comment:
@wangxiaoyou1993 Is this an acceptable place to put utility methods for the LogResource?

Reply:
Yes, you can put the utility methods here if they're not reused in other places.